IlmBase/config/CMakeLists.txt

```diff
@@ -78,7 +78,12 @@ if(ILMBASE_INSTALL_PKG_CONFIG)
     set(exec_prefix "\${prefix}")
     set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
     set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
-    set(LIB_SUFFIX_DASH ${ILMBASE_LIB_SUFFIX})
+    string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
+    if(uppercase_CMAKE_BUILD_TYPE MATCHES DEBUG)
```

Comment:
Why didn't you just compare to "Debug" which is always what the build type will be for debug?

Reply:
That's a good question. CMAKE_BUILD_TYPE is case insensitive, and a user could set it to debug, Debug, DEBUG, ... See for instance https://cmake.org/pipermail/cmake/2013-June/055177.html

Reply:
There's just as much danger from their calling it "Debugging" and not getting what they intended.  I think if we're worried about people setting the build type incorrectly, we should check early in the top-level CMakeLists.txt that CMAKE_BUILD_TYPE is set to one of the few choices we support. It seems error prone and overly verbose for every place it's referenced to need to separately handle for the possibility that somebody may have used nonstandard capitalization. 

I would advise just checking for "Debug" here, and as a separate PR we should put a check at the top level to make sure it's not set to anything other than "Release" or "Debug".

Reply:
Well, it is officially documented to be case insensitive (see https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html), but I've seen many projects do exactly what you say, so yes, can do.

Reply:
> I would advise just checking for "Debug" here, and as a separate PR we should put a check at the top level to make sure it's not set to anything other than "Release" or "Debug".

Strictly limiting the possible CMAKE_BUILD_TYPEs could lead to systems not being able to build a package. For example, on Gentoo, we use `CMAKE_BUILD_TYPE=Gentoo`, which is essentially a release build type, but ensures compiler and linker flags are set appropriately to the suers needs and the desired compiler and toolchain is used. Other source based distributions might have a similar way. However, I don't know how binary distributions handle this.

Reply:
Oh wow, I did not know that. Are there not a lot of projects that either (a) reject a build type they don't recognize, or (b) subtly do the wrong thing, like not turning on certain optimizations if the build type isn't "Release"?

Reply:
Neither of both are common. Some projects hardcode their build type, which is handled by commenting it out. If a project just overwrites eg CFLAGS without respecting user settings, we usually file a bug report to upstream, and ask about just adding their flags to the predefined ones. Maintainers usually also check the build scripts for pitfalls such as (b) and also report these upstream and seek a solution which works for both parties. See the bash script at https://github.com/gentoo/gentoo/blob/master/eclass/cmake.eclass#L277 if you want to know what is actually changed. We provide our own rules and toolchain files for building with cmake.

Reply:
Thanks for sharing your insight here @waebbl. I wasn't much aware of custom build types and their use in other projects. Release and Debug are definitely the most common ones, although I've seen RelWithDebInfo and MinSizeRel in the wild too, on occasion. Possible CMAKE_BUILD_TYPEs aside, is the request to amend this part of the patch (using a simpler case sensitive check against Debug) addressed to satisfaction?

Reply:
I don't know what the right fix here is. I think what @waebbl is saying is that somebody else could be using a totally different build type name to indicate a debug build, and that testing CMAKE_BUILD_TYPE here (even in a case-insensitive way) is not a reliable way to know if this is really a debug build or not?

Reply:
AFAIK the only reliable way to do this is to check case insensitively for a match with Debug, and also check if the user has requested a specific naming override, i.e. an additional optional flag such as OPENEXR_BUILD_TYPE to indicate that a Debug configuration is expected. That's the only pattern I've seen in other projects; use the "known to cmake" build types, Release, Debug, and RelWithDebInfo and MinSizeRel, *unless* MYPROJECT_BUILD_TYPE has been specified as an override, again with the cmake default names.

Reply:
BTW I noticed we don't set the _d suffix for Imath? We probably need to address that as well...

Reply:
> I don't know what the right fix here is. I think what @waebbl is saying is that somebody else could be using a totally different build type name to indicate a debug build, and that testing CMAKE_BUILD_TYPE here (even in a case-insensitive way) is not a reliable way to know if this is really a debug build or not?

That's actually not what I wanted to say. Althgouth this is possible, I think the chance is quite low. For example we always use the Gentoo build type and don't have a different one to indicate debug builds. What I wanted to say, is to try to avoid restricting a build to only the common four now types. Also if there's a need to add some special flags with a build type, try to append them to the `*{C,CXX,F}FLAGS` variables, so the users choices don't get overriden.
Many projects I've seen use a construct like in https://github.com/Kitware/VTK/blob/master/CMake/vtkInitializeBuildType.cmake which works great.

As for the issue at hand, the patch looks like a good way to solve it, but I don't know how debug builds are usually handled by pkg-config.

Reply:
> I don't know how debug builds are usually handled by pkg-config.

Debug builds have separate .pc file from release builds, to handle differences in naming schemes. In vcpkg for instance, the debug .pc files are installed in <prefix>/debug/lib/pkgconfig (along with the libraries in <prefix>/debug/lib) whilst the release .pc files are found in <prefix>/lib/pkgconfig (along with the libraries in <prefix>/lib).

> AFAIK the only reliable way to do this is to check case insensitively for a match with Debug,

That's my understanding too. If that's desired, I can revert https://github.com/AcademySoftwareFoundation/openexr/pull/1032/commits/ee02bdbb1138e825e662b48a3a5c35121ed6cdce (which I added to replace the case insensitive match with a case sensitive string equality, as I understood this was requested).

> and also check if the user has requested a specific naming override, i.e. an additional optional flag such as OPENEXR_BUILD_TYPE to indicate that a Debug configuration is expected.

That's a good idea.

Please let me know how you wish to proceed.

Reply:
Instead of generally discussing possible build types, I would suggest to focus on the particular case:
- openexr sets [`CMAKE_DEBUG_POSTFIX`](https://cmake.org/cmake/help/latest/variable/CMAKE_DEBUG_POSTFIX.html)
- "This variable is a special case of the more-general `CMAKE_<CONFIG>_POSTFIX` variable for the DEBUG configuration."
- As long as openexr sets `CMAKE_DEBUG_POSTFIX`, the DEBUG build type creates incorrect pc files. This is what this PR covers.

The cmake documentation implies that the user might also set `CMAKE_<CONFIG>_POSTFIX` for other build types. If you want correct pc files also in these cases, it needs more work. But it seems easy to handle them all without even knowing them in advance.

Reply:
Ah, yes, that's actually much cleaner, if possible. Is something like ``${CMAKE_${uppercase_CMAKE_BUILD_TYPE}_POSTFIX}`` guaranteed to cover all bases (defaulting to RELEASE if CMAKE_BUILD_TYPE is empty)?

Reply:
... I'm actually not sure if an empty build type implies release. The project could set a default CMAKE_BUILD_TYPE (and tell the user) if none is specified.